hudi-timeline-service/src/main/java/org/apache/hudi/timeline/service/RequestHandler.java

```diff
@@ -502,14 +502,20 @@ public void handle(@NotNull Context context) throws Exception {
         if (refreshCheck) {
           long beginFinalCheck = System.currentTimeMillis();
           if (isLocalViewBehind(context)) {
-            String errMsg =
-                "Last known instant from client was "
-                    + context.queryParam(RemoteHoodieTableFileSystemView.LAST_INSTANT_TS,
-                        HoodieTimeline.INVALID_INSTANT_TS)
-                    + " but server has the following timeline "
-                    + viewManager.getFileSystemView(context.queryParam(RemoteHoodieTableFileSystemView.BASEPATH_PARAM))
-                        .getTimeline().getInstants().collect(Collectors.toList());
-            throw new BadRequestResponse(errMsg);
+            String lastInstantTs = context.queryParam(RemoteHoodieTableFileSystemView.LAST_INSTANT_TS,
+                HoodieTimeline.INVALID_INSTANT_TS);
+            HoodieTimeline localTimeline =
+                viewManager.getFileSystemView(context.queryParam(RemoteHoodieTableFileSystemView.BASEPATH_PARAM)).getTimeline();
+            HoodieTimeline afterLastInstantTimeLine = localTimeline.findInstantsAfter(lastInstantTs).filterCompletedInstants();
+            if (!(afterLastInstantTimeLine.countInstants() == 1
```

Comment:
So, are we making an exception to just 1 case here. 
i.e. when client has 1 extra commit compared to timeline server and if that is a clean action, we don't trigger a refresh ? 
can you move this to a separate method. I see we might potentially add more cases in here going forward.

Reply:
`syncIfLocalViewBehind` is triggered before this. This judgment is added here mainly to avoid throwing exceptions during asynchronous clean, otherwise the program cannot execute normally.

I will try to move it to a separate method

Reply:
Done